tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc

```diff
@@ -726,6 +804,21 @@ StatusOr<std::vector<uint8>> CompileToHsaco(
 
   std::vector<uint8> hsaco;
   std::unique_ptr<llvm::TargetMachine> target_machine;
+  std::string ir_str;
+  llvm::raw_string_ostream stream(ir_str);
+  stream << *module;
+  std::string str = stream.str();
+  // Delete the first two lines, since they usually vary even when the rest of
+  // the code is the same (but verify that they are what we expect).
+  if (str.size() >= 13 && str.substr(0, 13) == "; ModuleID = ") {
```

Comment:
Would it make sense to use llvm string processing helpers here, or even llvm::Regex?

Reply:
The code does the job as written. Why bring in nonstandard APIs and add more complexity?